server/jamovi/core/columnw.cpp

```diff
@@ -117,31 +117,59 @@ void ColumnW::appendLevel(int value, const char *label)
 
     int length = strlen(label)+1;
     size_t allocated;
-
     char *chars = _mm->allocate<char>(length, &allocated);
-
     std::memcpy(chars, label, length);
 
+    length = strlen(importValue)+1;
+    size_t importAllocated;
+    char *importChars = _mm->allocate<char>(length, &importAllocated);
+    std::memcpy(importChars, importValue, length);
+
     s = struc();
     Level &l = _mm->resolve(s->levels)[s->levelsUsed];
 
     l.value = value;
     l.capacity = allocated;
     l.label = _mm->base(chars);
+    l.importCapacity = importAllocated;
+    l.importValue = _mm->base(importChars);
```

Comment:
120: space allocated for the label
125: space allocated for the import label
133: you convert the absolute location of the label to a relative one

note that if 125 results in the memory segment being moved, then the label pointer (from line 120) will be dangling, and 133 will seg fault.

when you have absolute pointers to shared memory, they can no longer be trusted once an allocation takes place - so you'd want to convert `chars` to the relative address (with `_mm->base()`) before you next call `_mm->allocate()`. to make matters worse, allocations happen quite rarely, making these behave correctly 99.9999% of the time.

but a lot of this will be solved if just give the labels and import values fixed sizes.